spec/extensions/forbid_lazy_load_spec.rb

```diff
@@ -15,312 +16,364 @@
-    proc{@o1.ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.mtm_ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.otoo_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.oto_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-  end
+    it "should forbid lazy load when using :forbid_lazy_load true association method option" do
+      proc{@o1.t(:forbid_lazy_load=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.ts(:forbid_lazy_load=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.mtm_ts(:forbid_lazy_load=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.otoo_t(:forbid_lazy_load=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.oto_t(:forbid_lazy_load=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+    end
 
-  it "should allow lazy load for instance if set at instance level" do
-    o = @c.all.first
-    o.allow_lazy_load.must_be_same_as o
-    o.t.must_equal @o2
-    o.ts.must_equal [@o2]
-    o.mtm_ts.must_equal [@o2]
-    o.otoo_t.must_equal @o2
-    o.oto_t.must_equal @o2
-  end
+    it "should forbid lazy load if set at instance level" do
+      @o1.forbid_lazy_load.must_be_same_as @o1
+      proc{@o1.t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.mtm_ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.otoo_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+      proc{@o1.oto_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+    end
 
-  it "should forbid lazy load if retrieved by dataset via Dataset#all" do
-    o = @c.all.first
-    proc{o.t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.otoo_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.mtm_ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.otoo_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.oto_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
+    it "should allow lazy load for instance if set at instance level" do
+      o = @c.all.first
+      o.allow_lazy_load.must_be_same_as o
+      o.t.must_equal @o2
+      o.ts.must_equal [@o2]
+      o.mtm_ts.must_equal [@o2]
+      o.otoo_t.must_equal @o2
+      o.oto_t.must_equal @o2
+    end
   end
 
-  it "should forbid lazy load if retrieved by dataset via Dataset#each" do
-    o = @c.each{|x| break x}
-    proc{o.t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.mtm_ts}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.otoo_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{o.oto_t}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-  end
+  describe "default options" do
```

Comment:
Most changes below are just the cases getting indented in by one so we could have two separate `describe` groups so `include model_instance_specs` works properly.